### ISSUE TITLE
Limit number of days displayed

### ISSUE THREAD

**Issue by space-ducks** [Leonezz/obsidian-tasks-calendar-wrapper#52]
Requesting an option to limit how far in advance is rendered by the plugin. I have some tasks that are still months out and obsidian becomes very sluggish when the plugin is trying to render all these days. If I hide the sidebar or close the plugin window it speeds back up. An option or default to limit to just a couple weeks out would be nice.

**Comment by Leonezz** [comment]
Hi @space-ducks thanks for the idea, it is reasonable to filter out outdated tasks and I will mark this as a new feature.

**Comment by MugenLee** [comment]
> Requesting an option to limit how far in advance is rendered by the plugin. I have some tasks that are still months out and obsidian becomes very sluggish when the plugin is trying to render all these days. If I hide the sidebar or close the plugin window it speeds back up. An option or default to limit to just a couple weeks out would be nice.



> Hi @space-ducks thanks for the idea, it is reasonable to filter out outdated tasks and I will mark this as a new feature.

Yes, when making annual plans, I will add tasks for the next few months, but currently I only want to focus on tasks for about one month (everyone is different)

By the way , some deep todo users may add different tags to different tasks, such as me. When displaying tasktimeline, you only want to display important todos, and it is a better choice to distinguish whether to display them through tags.

hope the tag filter can also be implemented soon

**Comment by Leonezz** [comment]
> > Requesting an option to limit how far in advance is rendered by the plugin. I have some tasks that are still months out and obsidian becomes very sluggish when the plugin is trying to render all these days. If I hide the sidebar or close the plugin window it speeds back up. An option or default to limit to just a couple weeks out would be nice.
> 
> > Hi @space-ducks thanks for the idea, it is reasonable to filter out outdated tasks and I will mark this as a new feature.
> 
> Yes, when making annual plans, I will add tasks for the next few months, but currently I only want to focus on tasks for about one month (everyone is different)
> 
> By the way , some deep todo users may add different tags to different tasks, such as me. When displaying tasktimeline, you only want to display important todos, and it is a better choice to distinguish whether to display them through tags.
> 
> hope the tag filter can also be implemented soon

About the tags filter, how do you like the form below if it could be implemented in this plugin:
![image](https://github.com/Leonezz/obsidian-tasks-calendar-wrapper/assets/33564074/5ead6a0b-253e-489e-b9b1-cb2194b0140b)

**Comment by MugenLee** [comment]
> > > Requesting an option to limit how far in advance is rendered by the plugin. I have some tasks that are still months out and obsidian becomes very sluggish when the plugin is trying to render all these days. If I hide the sidebar or close the plugin window it speeds back up. An option or default to limit to just a couple weeks out would be nice.
> > 
> > 
> > > Hi @space-ducks thanks for the idea, it is reasonable to filter out outdated tasks and I will mark this as a new feature.
> > 
> > 
> > Yes, when making annual plans, I will add tasks for the next few months, but currently I only want to focus on tasks for about one month (everyone is different)
> > By the way , some deep todo users may add different tags to different tasks, such as me. When displaying tasktimeline, you only want to display important todos, and it is a better choice to distinguish whether to display them through tags.
> > hope the tag filter can also be implemented soon
> 
> About the tags filter, how do you like the form below if it could be implemented in this plugin: ![image](https://private-user-images.githubusercontent.com/33564074/294837660-5ead6a0b-253e-489e-b9b1-cb2194b0140b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ3MjczNzUsIm5iZiI6MTcwNDcyNzA3NSwicGF0aCI6Ii8zMzU2NDA3NC8yOTQ4Mzc2NjAtNWVhZDZhMGItMjUzZS00ODllLWI5YjEtY2IyMTk0YjAxNDBiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA4VDE1MTc1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMyMWJmZTFlMzc0MmM2OTIxZTY4MjA3OWE0NWQzNzg3Y2FlNGU5YjhjZGNjOTYwMmE5Y2MyODA2OTcwZDg1NDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ppB9yGOxYrVyS4OdgG4B5Kcp-ImHIacUn9R3yIkOlOY)

This shouldn't be a high-frequency operation, so I think it can be done in the plugin settings interface.
I think it is: 
There is a tag toggle. After it is turned on, take some tags, and then the tasks with these tags will be displayed.Tasks that do not contain these tags will not be displayed
Because some unimportant tasks are often not tagged, but are only used as todos.
Here are some of my thoughts, hope they help you
